game-core/src/main/java/games/strategy/triplea/delegate/ExecutionStack.java

```diff
@@ -28,37 +29,34 @@
  */
 public class ExecutionStack implements Serializable {
   private static final long serialVersionUID = -8675285470515074530L;
-  private IExecutable m_current;
-  @SuppressWarnings("JdkObsolete") // change to Deque/ArrayDeque upon next incompatible release
-  private final Stack<IExecutable> m_stack = new Stack<>();
+  private IExecutable currentStep;
+  private final Deque<IExecutable> deque = new ArrayDeque<>();
 
   void execute(final IDelegateBridge bridge) {
     // we were interrupted before, resume where we left off
-    if (m_current != null) {
-      m_stack.push(m_current);
+    if (currentStep != null) {
+      currentStep.execute(this, bridge);
```

Comment:
Is this change intentional?

Reply:
It is. 
I figured it won't alter the behaviour as this item is instantly popped off the stack again and reassigned to this field again.